omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java

```diff
@@ -910,7 +894,65 @@ private void addPaths() {
 		Collections.sort(tagList);
 		swaggerSpecification.setTags(tagList);
 	}
-	
+
+	/**
+	 * Extracts resource name from a resource handler
+	 * @param trim if false returns resource name with version, if true version part of the name is trimmed
+	 * @return resource name
+	 */
+	private String getResourceName(DelegatingResourceHandler resourceHandler, boolean trim) {
+		Resource annotation = resourceHandler.getClass().getAnnotation(Resource.class);
+
+		String resourceName = null;
+		if (annotation != null) {
+			// top level resource
+			resourceName = annotation.name();
+		} else {
+			// sub resource
+			SubResource subResourceAnnotation = resourceHandler.getClass().getAnnotation(SubResource.class);
+			if (subResourceAnnotation != null) {
+				resourceName = subResourceAnnotation.path();
+			}
+		}
+		if (trim) {
+			resourceName = trimResourceName(resourceName);
+		}
+		return resourceName;
+	}
+
+	/**
+	 * Extracts resource parent name from a resource handler
+	 * @param trim if false returns name with version, if true version part of the name is trimmed
+	 * @return name of the parent resource or null if its not a sub-resource
+	 */
+	private String getResourceParentName(DelegatingResourceHandler resourceHandler, boolean trim) {
+		SubResource subResourceAnnotation = resourceHandler.getClass().getAnnotation(SubResource.class);
```

Comment:
There should be a way to minimize this duplication by making this method delegate to the one above

Reply:
@gayanW now that RESTWS-562 is merged, is it time to revisit this?

Reply:
With the final changes made to RESTWS-562, these methods will not be reused. So I think, we better not add more code. @dkayiwa @wluyima Sorry for all the time that it took to review. :) 

May be we should close the ticket or move it to backlog?

Reply:
Feel free to close it.